Create help-wanted.md

kvncrw opened the pull request:
initial commit of help-wanted framework

- introduce a guide intended to encourage contributors by specifically labeling issues we feel are good for first-time contributors `contributor/good-first-issue` AND
- introduce a label that indicates the issue is groomed enough for regular contributors and external maintainers to work on independently `contributor/help-wanted`

internal discussions should also take place on how we want to handle the management and grooming of these issues, particularly the cadence, priority, and ownership of these tasks.

kvncrw committed:
Create help-wanted.md

initial commit of help-wanted framework

notsureifkevin commented:
This needs to be formatted using https://sembr.org/

kvncrw committed:
Update help-wanted.md

sembr

jbdoumenjou commented on help-wanted.md:
```diff
@@ -0,0 +1,126 @@
+# Overview
```
As a general comment, I would apply the Keep It Simple Stupid principle.

Instead of trying to define a specific process with a list of constraints/LiveCycle, I would prefer to welcome contributors with a message that gives confidence and some specifics references to start (cf next comments).

One advantage of a message when we apply the label is that the contributor has all the needed information to start to work and doesn't have to be afraid of an external reference or specifics rules to apply. 
The other advantage is to focus on the contributor experience and adapt our behavior to have a personalized experience.

jbdoumenjou commented on help-wanted.md:
```diff
@@ -0,0 +1,126 @@
+
+We also have some suggestions for using these labels to help grow and improve our community.
+
+## Help Wanted
```
For me, this label is for a wanted item but we don't have time to work on it.
The only constraint on it is that we considered that an external contributor could work on it whatever the complexity/context of the task.

We can decide to ask for help whatever the difficulty or context and we can answer with a message that provides the needed information to start to work on it and give confidence.

rtribotte replied:
I agree with @jbdoumenjou, as long as we are choosing those issues, the complexity should not be strict criteria, but an element of discussion during the labeling an issue

kvncrw replied:
> We can decide to ask for help whatever the difficulty or context and we can answer with a message that provides the needed information to start to work on it and give confidence.

The idea behind meeting the criteria listed below beforehand is to provide contributors a path to start investigating the issue by being empowered with clear guidelines and reference materials. I understand that puts additional work on the maintainers upfront, but the tradeoff is that the barrier to entry on recruiting contributors is lowered. I'm not opposed to deferring this work until a contributor indicates they're willing to work on it, but would the team be open to A/B testing this?

Perhaps we identify 6 issues, tag them as `help-wanted`, and work on providing some of the details outlined below for 3 of them. Then I will track all 6 of them and measure a few key metrics:

* Time to first claim
* Frequency of comments / communication in the issue / subsequent PR
* Time to PR opened
* Time to PR closed

jbdoumenjou commented on help-wanted.md:
```diff
@@ -0,0 +1,126 @@
+    If the area of code doesn't have tests, before labeling the issue, add a test fixture.
+    This prep often makes a great `contributor/good-first-issue` task!
+
+# Suggestions
```
All of this part concerns communication management and does not seem relevant here.

kvncrw replied:
I'm still recovering from illness, so I'm still a bit hazy -- I'll probably hold off on conversating on the other points because I'm unclear on if there are any action items for them.

However, for this one, I believe the goal is to provide communication management advice for both internal AND  external maintainers and contributors. We could put this somewhere in notion, or elsewhere, but it seems relevant here if our goal is to eventually empower members of the community to step up and engage in these activities. WDYT?

jbdoumenjou commented on help-wanted.md:
```diff
@@ -0,0 +1,126 @@
+    are no longer desired, no longer make sense, have changed priority or difficulty, etc.
+    The maintainers of Traefik regularly groom and manage these issues to ensure they remain relevant.
+
+## Good First Issue
```
In the same way, I would just add a message when we apply the label with something like "We will have special attention and help you with this PR/issue, you can find some doc here and here, don't hesitate to ask for help".
And avoid writing another external document to read.

I think we don't have to apply any process/LiveCycle that takes the number of `good-first-issue` PR into account to know if someone can work on a `contributor/help-wanted`. For me `help-wanted` and `good-first-issue` are not correlated and we can't know the skills nor the energy that the contributor could/would provide.

kvncrw replied:
> In the same way, I would just add a message when we apply the label with something like "We will have special attention and help you with this PR/issue, you can find some doc here and here, don't hesitate to ask for help".
And avoid writing another external document to read.

That's what the goal of the guidelines below are. That we prepare the issue accordingly to encourage new contributors and empower them with the details they need to start contributing. We should definitely timebox this effort, and be aware that not all these suggestions can always be met, but I would encourage the team to try and think about the bare minimum that is needed for new contributors to get rolling on an issue. (cf previous comment on help-wanted applies here as well)

kvncrw commented on help-wanted.md:
```diff
@@ -0,0 +1,126 @@
+what's the acceptable way to ask for people to review a PR,
+nudge things along when a PR is stalled. Show them how we operate
+by helping move their first PR along.
+- If you have time, let the contributor know that they can DM you on the Ambassador discord
```
take this out

kvncrw committed:
minor cleanup

- renamed to be consistent with other files in this project
- removed ambassador stuff
- added clarity around help wanted / good first issue pre-requisites

rtribotte reviewed (approved):
LGTM 👌

kevinpollet reviewed (approved):
LGTM 👍

jspdown reviewed (approved):
LGTM
